app/src/main/java/com/limelight/binding/input/ControllerHandler.java

```diff
@@ -827,6 +831,22 @@ private int handleRemapping(InputDeviceContext context, KeyEvent event) {
             }
         }
 
+        // Switch A with B and X with Y button for 8BitDo controllers
+        if (context.is8BitDoBtController) {
```

Comment:
Now that we have generic support for flipping face buttons, all we need for 8BitDo is the mapping for the mode button.

I think you can drop all your other changes except for something like:
```
// Some comment here
if (context.vendorId == 0x2dc8 && event.getScanCode() == 306) {
    return KeyEvent.KEYCODE_BUTTON_MODE;
```

Reply:
Ok makes sense. My code is probably somehow redundant. Will change that.